pcsx2-qt/MainWindow.cpp

```diff
@@ -1014,6 +1025,85 @@ void MainWindow::onLoggingOptionChanged()
 	QtHost::UpdateLogging();
 }
 
+void MainWindow::onInputRecNewActionTriggered()
+{
+	const bool wasPaused = VMManager::GetState() == VMState::Paused;
+	const bool wasRunning = VMManager::GetState() == VMState::Running;
+	if (wasRunning && !wasPaused)
+	{
+		VMManager::SetPaused(true);
+	}
+
+	NewInputRecordingDlg dlg(this);
+	const auto result = dlg.exec();
+
+	if (result == QDialog::Accepted)
+	{
+		if (g_InputRecording.Create(
+				dlg.getFilePath(),
+				dlg.getInputRecType() == InputRecording::Type::FROM_SAVESTATE,
+				dlg.getAuthorName()))
+		{
+			return;
+		}
+	}
+
+	if (wasRunning && !wasPaused)
+	{
+		VMManager::SetPaused(false);
+	}
+}
+
+#include "pcsx2/Recording/InputRecordingControls.h"
+
+void MainWindow::onInputRecPlayActionTriggered()
+{
+	const bool wasPaused = VMManager::GetState() == VMState::Paused;
+
+	if (!wasPaused)
+		g_InputRecordingControls.PauseImmediately();
```

Comment:
Not sure what this function does exactly, but I saw it being manipulated on the emu thread as well. If you need to call it from the UI thread, it should go through `EmuThread`, with a queued connection slot (see the other methods).

Reply:
I'll keep that in mind when I refactor the class, once again just replicating existing behaviour for this initial PR.

Reply:
There's `Host::RunOnCPUThread()` which may be helpful here, but I never got around to implementing it in Qt. Will do that at some point, as well as the inverse (`RunOnUIThread()`).